.github/workflows/main.yml

```diff
@@ -38,10 +36,14 @@ jobs:
 
   release:
     if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
-    needs: test
+    needs: 
+      - test
+      - check
     runs-on: ubuntu-latest
     permissions:
       id-token: write
+      contents: write
+      pull-requests: write
```

Comment:
hoping it fix like https://github.com/google-github-actions/release-please-action/issues/709